agent/fw_yii.c

```diff
@@ -86,21 +87,141 @@ NR_PHP_WRAPPER(nr_yii_runWithParams_wrapper) {
+      NR_PSTR("CAction::runWithParams"), nr_yii1_runWithParams_wrapper, NULL,
       NULL);
   nr_php_wrap_user_function_before_after_clean(
-      NR_PSTR("CInlineAction::runWithParams"), nr_yii_runWithParams_wrapper,
+      NR_PSTR("CInlineAction::runWithParams"), nr_yii1_runWithParams_wrapper,
       NULL, NULL);
 #else
   nr_php_wrap_user_function(NR_PSTR("CAction::runWithParams"),
-                            nr_yii_runWithParams_wrapper TSRMLS_CC);
+                            nr_yii1_runWithParams_wrapper TSRMLS_CC);
   nr_php_wrap_user_function(NR_PSTR("CInlineAction::runWithParams"),
-                            nr_yii_runWithParams_wrapper TSRMLS_CC);
+                            nr_yii1_runWithParams_wrapper TSRMLS_CC);
+#endif
+}
+
+/*
+ * Yii2: Set the web transaction name from the unique action ID.
+ */
+NR_PHP_WRAPPER(nr_yii2_runWithParams_wrapper) {
+  zval* this_var = NULL;
+  zval* unique_idz = NULL;
+  const char* unique_id = NULL;
+  nr_string_len_t unique_id_length;
+  char* transaction_name = NULL;
+
+  (void)wraprec;
+  NR_UNUSED_SPECIALFN;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_YII2);
+
+  this_var = nr_php_scope_get(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  if (NULL == this_var) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "Yii2: improper this");
+    goto end;
+  }
+
+  unique_idz = nr_php_call(this_var, "getUniqueId");
+  if (nr_php_is_zval_non_empty_string(unique_idz)) {
+    unique_id = Z_STRVAL_P(unique_idz);
+    unique_id_length = Z_STRLEN_P(unique_idz);
+
+    if (unique_id_length > 256) {
+      nrl_warning(NRL_FRAMEWORK,
+                  "Yii2 unique ID is too long (> %d); Yii2 naming not used",
+                  256);
+    } else {
+      transaction_name = (char*)nr_alloca(unique_id_length + 1);
+      nr_strxcpy(transaction_name, unique_id, unique_id_length);
+
+      nr_txn_set_path("Yii2", NRPRG(txn), transaction_name, NR_PATH_TYPE_ACTION,
+                      NR_NOT_OK_TO_OVERWRITE);
+    }
+  } else {
+    nrl_verbosedebug(NRL_FRAMEWORK,
+                     "getUniqueId does not return a non empty string (%d)",
+                     Z_TYPE_P(unique_idz));
+  }
+  nr_php_zval_free(&unique_idz);
+end:
+  NR_PHP_WRAPPER_CALL;
+
+  nr_php_scope_release(&this_var);
+}
+NR_PHP_WRAPPER_END
+
+#if ZEND_MODULE_API_NO < ZEND_8_0_X_API_NO \
+    || defined OVERWRITE_ZEND_EXECUTE_DATA
+/*
+ * Yii2: Report errors and exceptions when built-in ErrorHandler is enabled.
+ */
+NR_PHP_WRAPPER(nr_yii2_error_handler_wrapper) {
+  zval* exception = NULL;
+
+  NR_UNUSED_SPECIALFN;
+  (void)wraprec;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_YII2);
+
+  exception = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  if (NULL == exception || !nr_php_is_zval_valid_object(exception)) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: exception is NULL or not an object",
+                     __func__);
+    NR_PHP_WRAPPER_CALL;
+    goto end;
+  }
+
+  NR_PHP_WRAPPER_CALL;
+
+  if (NR_SUCCESS
```

Comment:
why does this need to happen after NR_PHP_WRAPPER_CALL. We don't need anything else that we get from running the function in wrapper_call. Can't we simplify the whole thing.
something like:

```
  exception = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
  if (NULL == exception || !nr_php_is_zval_valid_object(exception)) {
    nrl_verbosedebug(NRL_FRAMEWORK, "%s: exception is NULL or not an object",
                     __func__);
  } else {
  if (NR_SUCCESS
!= nr_php_error_record_exception(
          NRPRG(txn), exception, nr_php_error_get_priority(E_ERROR), true,
          "Uncaught exception ", &NRPRG(exception_filters) TSRMLS_CC)) {
    nrl_verbosedebug(NRL_FRAMEWORK, "%s: unable to record exception", __func__);
  }
}
  nr_php_arg_release(&exception);
NR_PHP_WRAPPER_CALL;
}
NR_PHP_WRAPPER_END

Reply:
0b892a6211096b0bd3f716b30f9462e83a8c38e3?